apps/google-analytics-4/frontend/src/components/config-screen/assign-content-type/AssignContentTypeRow.tsx

```diff
@@ -24,44 +29,98 @@ const AssignContentTypeRow = (props: Props) => {
     onContentTypeChange,
     onContentTypeFieldChange,
     onRemoveContentType,
+    currentEditorInterface,
+    originalParameters,
   } = props;
 
   const [contentTypeId, { slugField, urlPrefix }] = contentTypeEntry;
 
-  const renderStatusItem = (key: string, slugField: string) => {
-    if (key && !slugField) {
-      return (
-        <Box className={styles.statusItem} testId="warningIcon">
-          <Tooltip content="This content type must have a slug field selected in order for the app to render correctly in the sidebar">
-            <WarningIcon variant="warning" />
-          </Tooltip>
-        </Box>
-      );
+  const [isSaved, setIsSaved] = useState<boolean>(false);
+  const [contentTypeOptions, setContentTypeOptions] = useState<AllContentTypeEntries>([]);
+  const [isContentTypeInOptions, setIsContentTypeInOptions] = useState<boolean>(true);
+  const [isSlugFieldInOptions, setIsSlugFieldInOptions] = useState<boolean>(true);
+  const [isInSidebar, setIsInSidebar] = useState<boolean>(false);
+
+  useEffect(() => {
+    const originalContentTypeIds = Object.keys(originalParameters.contentTypes);
+    if (originalContentTypeIds.includes(contentTypeId)) {
+      setIsSaved(true);
+    } else {
+      setIsSaved(false);
+    }
+  }, [contentTypeId, originalParameters]);
+
+  useEffect(() => {
+    const savedSidebarLocations = Object.keys(currentEditorInterface);
+    if (savedSidebarLocations.includes(contentTypeId)) {
+      setIsInSidebar(true);
     } else {
-      return <Box className={styles.statusItem} testId="noStatus" />;
+      setIsInSidebar(false);
+    }
+  }, [contentTypeId, currentEditorInterface]);
+
+  useEffect(() => {
+    const getContentTypeOptions = () => {
+      const contentTypeOptions = allContentTypeEntries.filter(
+        ([type]) => type === contentTypeId || !contentTypes[type]
+      );
+      setContentTypeOptions(contentTypeOptions);
+
+      if (isSaved) {
+        setIsContentTypeInOptions(
+          contentTypeOptions.find((option) => option[0] === contentTypeId) ? true : false
+        );
+        if (slugField !== undefined) {
+          setIsSlugFieldInOptions(
+            allContentTypes[contentTypeId]?.fields.find((field) => field.id === slugField)
+              ? true
+              : false
+          );
+        }
+      }
+    };
+
+    getContentTypeOptions();
```

Comment:
This isn't an async function so we don't need to do wrap the logic in a separate function and call it ... we can just stick it at the main useEffect scope.